lua/neogit/popups/branch/actions.lua

```diff
@@ -22,6 +22,60 @@ local function parse_remote_branch_name(ref)
   return remote, branch_name
 end
 
+M.spin_off_branch = operation("spin_off_branch", function()
+  local name = input.get_user_input("branch > ")
+  if not name or name == "" then
+    return
+  end
+
+  name, _ = name:gsub("%s", "-")
+  git.branch.create(name)
+
+  local current_branch_name = git.branch.current_full_name()
+
+  git.cli.checkout.branch(name).call_sync()
+
+  local upstream = git.branch.upstream()
+  if upstream then
+    git.log.update_ref(current_branch_name, upstream)
+  end
+
+  status.refresh(true, "spin_off_branch")
+end)
+
+M.spin_out_branch = operation("spin_out_branch", function()
+  local checkout = false
+  if status.is_dirty() then
+    notif.create("Staying on HEAD due to uncommitted changes", vim.log.levels.INFO)
+    checkout = true
+  end
+
+  local name = input.get_user_input("branch > ")
+  if not name or name == "" then
+    return
+  end
+
+  name, _ = name:gsub("%s", "-")
```

Comment:
Oh, thats something I added a while back. Git branches cant include a space, so instead of erroring, i opted to fix it for the user. IMO its super handy.